lib/encryption/key_manager.dart

```diff
@@ -552,7 +555,8 @@ class KeyManager {
 
     final deviceKeys = await room.getUserDeviceKeys();
     final deviceKeyIds = _getDeviceKeyIdMap(deviceKeys);
-    deviceKeys.removeWhere((k) => !k.encryptToDevice);
+    deviceKeys
+        .removeWhere((k) => !k.encryptToDevice && k.userId == client.userID!);
```

Comment:
Shouldn't this be an OR? 🤔 We want to remove devices where `device.encryptToDevice` is `false` OR the device is not from the own user

Reply:
Also same here. If we don't share the outbound group session with other users, we can not communicate with them at all